Add new Xjit option enableAggressiveLoopVersioning

JamesKingdon opened the pull request:
enableAggressiveLoopVersioning is a new jit option that will be used to
guard a number of changes to the behaviour of LoopVersioner.

Signed-off-by: JamesKingdon <jkingdon@ca.ibm.com>

JamesKingdon commented:
This will be a pre-req for four upcoming changes.

fjeremic commented on compiler/control/OMROptions.cpp:
```diff
@@ -100,6 +100,7 @@ TR::OptionTable OMR::Options::_jitOptions[] = {
         TR::Options::setStaticNumeric, (intptrj_t)&OMR::Options::_abstractTimeToReduceInliningAggressiveness, 0, "F%d", NOT_IN_SUBSET },
    {"acceptHugeMethods",     "O\tallow processing of really large methods", SET_OPTION_BIT(TR_ProcessHugeMethods), "F" },
    {"activateCompThreadWhenHighPriReqIsBlocked", "M\tactivate another compilation thread when high priority request is blocked",  SET_OPTION_BIT(TR_ActivateCompThreadWhenHighPriReqIsBlocked), "F", NOT_IN_SUBSET},
+   {"aggressiveLoopVersionerOpts", "O\tUse more loop versioning", SET_OPTION_BIT(TR_aggressiveLoopVersionerOpts), "F" },
```
This option sounds weird because its not in an imperative mood mood. IMO options should be commands. Would `aggressivelyVersonLoops` not sounds more imperative?

JamesKingdon replied:
We have several precedents for non-imperative options, and I wasn't aware of an intent to make Xjit options follow an imperative naming scheme. Isn't it reasonable to consider any boolean option foo to be implicitly equivalent to enableFoo, with the 'enable' omitted for brevity?

I have no particular attachment to the name though, so happy to change it if you wish.

fjeremic replied:
@JamesKingdon this was just _my_ personal preference and as such I did not request changes in my review, but rather a 'review comment' in GitHub terms. To explain my rationale to me `-Xjit:aggressiveLoopVersionerOpts` doesn't really say much. Are we enabling aggressive loop versioner opts? Are we tracing them? Are we disabling them?

fjeremic replied:
I've created https://github.com/eclipse/omr/issues/1370 to kick things off.

JamesKingdon replied:
Excellent, that looks like a useful discussion to have. I'll push forward on the dependent changes and circle round to the name for the option at the end, by which time I can take into account any feedback from #1370

JamesKingdon commented:
I changed the name of the option to take into account the comments from Filip and #1370, and rebased to resolve a conflict on the choice of bit to represent the option.

vijaysun-omr commented:
"aggressiveLoopVersionerOpts" is not the name of the option but is mentioned in the PR description ?
